Rakefile

```diff
@@ -602,14 +602,51 @@ namespace :changelog do
   end
 end
 
-Rake::ExtensionTask.new("ddtrace_profiling_native_extension.#{RUBY_VERSION}_#{RUBY_PLATFORM}") do |ext|
+native_exts = []
+
+native_exts << Rake::ExtensionTask.new("ddtrace_profiling_native_extension.#{RUBY_VERSION}_#{RUBY_PLATFORM}") do |ext|
   ext.ext_dir = 'ext/ddtrace_profiling_native_extension'
 end
 
-Rake::ExtensionTask.new("ddtrace_profiling_loader.#{RUBY_VERSION}_#{RUBY_PLATFORM}") do |ext|
+native_exts << Rake::ExtensionTask.new("ddtrace_profiling_loader.#{RUBY_VERSION}_#{RUBY_PLATFORM}") do |ext|
   ext.ext_dir = 'ext/ddtrace_profiling_loader'
 end
 
+NATIVE_CLEAN = ::Rake::FileList[]
+namespace :native_dev do
+  compile_commands_tasks = native_exts.map do |ext|
+    tmp_dir_dd_native_dev = "#{ext.tmp_dir}/dd_native_dev"
+    directory tmp_dir_dd_native_dev
+    NATIVE_CLEAN << tmp_dir_dd_native_dev
+
+    compile_commands_task = file "#{ext.ext_dir}/compile_commands.json" => [tmp_dir_dd_native_dev] do |t|
```

Comment:
I guess one downside of this approach is that it works only for a specific Ruby version, so if we need to switch around, we'll need to re-run the rake task every time (and remember to do so).

On the other hand, I think that's fine ATM -- I'm guessing it may be non-trivial to make this per-ruby-version.

Reply:
Yeah I think doing some automation here may require either encoding the switch operation as a rake task or formally aligning on an official way to handle different Ruby versions (e.g. asdf) so we could use some kind of hook system.

Both options sound way too complicated for the benefit atm 😅